### PR TITLE
Add a check for file writability to the DB to confuse users slightly less.

### PR DESCRIPTION
**Pull request by t-h-i-s-u-s-e-r-n-a-m-e-i-s-c-a-n-c-e-r**
I didn't find a Qt way of doing this quickly enough so I used access(). Hope that's OK.

### CODE REVIEW

**Review by oldmud0 — CHANGES_REQUESTED** [review]
Read the docs. Use QFileInfo::permission.

**Comment by t-h-i-s-u-s-e-r-n-a-m-e-i-s-c-a-n-c-e-r** [comment]
Sure, but I'd need to check if we're the owner or in the owners group, just seems unnecessary to me.

File: core/src/db_manager.cpp [hunk 1]
```diff
@@ -20,8 +20,12 @@
 DBManager::DBManager() :
     DRIVER("QSQLITE")
 {
-    if(!access("config/akashi.db", R_OK | W_OK))
+    QFileInfo *db_info = new QFileInfo;
```
Comment (Salanto) [inline core/src/db_manager.cpp:23]: Wouldn't a local variable work here too? Why a pointer?

File: core/src/db_manager.cpp [hunk 2]
```diff
@@ -20,6 +20,13 @@
 DBManager::DBManager() :
     DRIVER("QSQLITE")
 {
+    QFileInfo *db_info = new QFileInfo;
+    *db_info = QFileInfo("config/akashi.db");
+    if(!(db_info->isReadable()) || !(db_info->isWritable()))
+        qCritical() << "Database Error: Missing permissions. Check if \"config/akashi.db\" is writable.\n";
+    delete db_info;
+    
+
```
Comment (oldmud0) [inline core/src/db_manager.cpp:29]:
```suggestion
    const QString db_filename = "config/akashi.db";
    QFileInfo db_info(db_filename);
    if(!db_info.isReadable() || !db_info.isWritable()) 
        qCritical() << tr("Database Error: Missing permissions. Check if \"%1\" is writable.").arg(db_filename);

```

**Review by oldmud0 — APPROVED** [review]

**Pull request merged**